build-time-analytics/src/test/java/io/quarkus/ts/buildtimeanalytics/AbstractAnalyticsIT.java

```diff
@@ -0,0 +1,139 @@
+package io.quarkus.ts.buildtimeanalytics;
+
+import static io.quarkus.ts.buildtimeanalytics.AnalyticsFilesUtils.getAnalyticsPayloadFile;
+import static io.quarkus.ts.buildtimeanalytics.AnalyticsFilesUtils.getPomFile;
+import static io.quarkus.ts.buildtimeanalytics.AnalyticsUtils.QUARKUS_ANALYTICS_FAKE_URI_BASE;
+import static io.quarkus.ts.buildtimeanalytics.AnalyticsUtils.QUARKUS_ANALYTICS_IP;
+import static io.quarkus.ts.buildtimeanalytics.AnalyticsUtils.QUARKUS_ANALYTICS_URI_BASE_PROPERTY;
+import static io.quarkus.ts.buildtimeanalytics.AnalyticsUtils.UNRECOGNIZED_PROPERTY_FORMAT;
+import static io.quarkus.ts.buildtimeanalytics.AnalyticsUtils.getCreateAppRequest;
+import static io.quarkus.ts.buildtimeanalytics.AnalyticsUtils.getExtensionVersionPattern;
+import static io.quarkus.ts.buildtimeanalytics.AnalyticsUtils.getNonExcludedAppName;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+import java.io.File;
+import java.io.FileReader;
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Set;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+
+import jakarta.inject.Inject;
+
+import org.apache.maven.model.Dependency;
+import org.apache.maven.model.Model;
+import org.apache.maven.model.io.xpp3.MavenXpp3Reader;
+import org.codehaus.plexus.util.xml.pull.XmlPullParserException;
+import org.hamcrest.Matchers;
+
+import io.quarkus.test.bootstrap.QuarkusCliClient;
+import io.quarkus.test.bootstrap.QuarkusCliClient.Result;
+import io.quarkus.test.bootstrap.QuarkusCliRestService;
+import io.quarkus.test.services.quarkus.model.QuarkusProperties;
+import io.restassured.path.json.JsonPath;
+
+public abstract class AbstractAnalyticsIT {
+    @Inject
+    static QuarkusCliClient cliClient;
+
+    protected QuarkusCliRestService whenCreated() {
```

Comment:
What do you mean by `assertions should be inside test classes`?

Dropping `when` makes total sense here, as the methods have return values.

I'd like to keep the `then` methods, because otherwise there would be many dupicities. How about dropping the `then` in their names as well? It would be harder to tell them apart, though.

Reply:
I mean, that all calls of eg this method:
```
   protected void thenBuildSuccessful(Result buildResult) {
        assertTrue(buildResult.isSuccessful(), "The application didn't build on JVM. Output: " + buildResult.getOutput());
    }
```
can be replaced with
```
assertTrue(buildResult.isSuccessful(), "The application didn't build on JVM. Output: " + buildResult.getOutput());
```

for methods, which you like to keep, I suggest to use verb "validate", "verify", "check", "ensure" or even "is".

Reply:
Ok, I'll rename
```
when<executed> -> <execute>
then* -> verify*
```
But I'd like to keep most of the `veirfy` methods and not inline them as you suggest. To me, it makes no sense to repeat the exact same `assertTrue(...)` statement 10-times.